packages/bierzo-wallet/src/store/accounts/reducer.ts

```diff
@@ -0,0 +1,38 @@
+import { ChainAddressPair } from "@iov/bns";
+import { Action } from "redux";
+import { ActionType } from "typesafe-actions";
+
+import * as actions from "./actions";
+
+export interface BwAccount {
+  readonly name: string;
+  readonly domain: string;
+  readonly expiryDate: Date;
+  readonly addresses: readonly ChainAddressPair[];
+}
+
+export interface AddAccountsActionType extends Action {
+  readonly type: "@@accounts/ADD";
+  readonly payload: readonly BwAccount[];
+}
+
+export type AccountsActions = ActionType<typeof actions>;
+
+export type AccountsState = readonly BwAccount[];
+const initState: AccountsState = [];
+
+export function accountsReducer(state: AccountsState = initState, action: AccountsActions): AccountsState {
+  switch (action.type) {
+    case "@@accounts/ADD": {
+      const updatedAccounts = action.payload.map(
+        updatedAccount => `${updatedAccount.name}*${updatedAccount.domain}`,
+      );
+      const oldAccountsToCopy = state.filter(
+        oldAccount => !updatedAccounts.includes(`${oldAccount.name}*${oldAccount.domain}`),
+      );
+      return [...oldAccountsToCopy, ...action.payload];
```

Comment:
To be honest I do not understand why you are `Updating` state in `Add` reducer. Reducer should be simple enough and contain only one state modification. Also please make sure that you are sorting records inside array on component side or on reducer side to avoid jump of the updated accounts to the end of the list of accounts.

Reply:
I followed the same approach as Simon followed when implementing username's store [here](https://github.com/iov-one/ponferrada/commit/57990e5353c2aa633523e1df7e9e993132fd543c#diff-8b0f2a8038ca75e3efd2da744d95eab0). The idea is to make a partial replace with new the accounts.

Reply:
Ok then